core/src/main/java/org/apache/iceberg/util/PropertyUtil.java

```diff
@@ -70,4 +70,45 @@ public static String propertyAsString(Map<String, String> properties,
     }
     return defaultValue;
   }
+
+  @FunctionalInterface
+  private interface PropertyResolver<A, B, C, R> {
+    R apply(A argA, B argB, C argC);
+  }
+
+  private static <T> T resolveProperty(PropertyResolver<Map<String, String>, String, T, T> resolver,
+                                    Map<String, String> options, String optionKey,
+                                    Map<String, String> properties, String propertyKey, T defaultVal) {
+    return resolver.apply(options, optionKey, resolver.apply(properties, propertyKey, defaultVal));
+  }
+
+  private static <T> T resolveProperty(PropertyResolver<Map<String, String>, String, T, T> resolver,
+                                      Map<String, String> options, String optionKey,
```

Comment:
Are there better names than options, properties1, and properties2? How about `priorityOneProps`, `priorityTwoProps` etc.? That's more readable.

Reply:
I think what Russell suggests for a `SparkActionsConf` sounds like a better way, let me update the PR in that way and see how that goes